### ISSUE TITLE
Rule idea: require a placeholder when using @extend

### ISSUE THREAD

**Issue by kristerkari** [kristerkari/stylelint-scss#1]
Sometimes you only use `@extend` to share code between two or more classes, but never actually use the class that your extending. In this case it would be a good idea to require `@extend` to only be used with placeholder selectors (e.g. `%my-class`).

For example, this code:

```scss
.icon {
  transition: background-color ease .2s;
  margin: 0 .5em;
}

.error-icon {
  @extend .icon;
  /* error specific styles... */
}

.info-icon {
  @extend .icon;
  /* info specific styles... */
}
```

...gets compiled to:

```scss
.icon, .error-icon, .info-icon {
  transition: background-color ease .2s;
  margin: 0 .5em;
}

.error-icon {
  /* error specific styles... */
}

.info-icon {
  /* info specific styles... */
}
```

You have `.icon` defined in your CSS even if you don't plan to use it.

However, if you use a placeholder, this:

```scss
%icon {
  transition: background-color ease .2s;
  margin: 0 .5em;
}

.error-icon {
  @extend %icon;
  /* error specific styles... */
}

.info-icon {
  @extend %icon;
  /* info specific styles... */
}
```

...gets compiled to:

```scss
.error-icon, .info-icon {
  transition: background-color ease .2s;
  margin: 0 .5em;
}

.error-icon {
  /* error specific styles... */
}

.info-icon {
  /* info specific styles... */
}
```

All examples taken from:
http://thesassway.com/intermediate/understanding-placeholder-selectors
